library/src/G4CMPUtils.cc

```diff
@@ -199,12 +216,25 @@ G4ThreeVector G4CMP::LambertReflection(const G4ThreeVector& surfNorm) {
 
 
 // Check that phonon is properly directed from the volume surface
+// waveVector and surfNorm need to be in global coordinates
 
 G4bool G4CMP::PhononVelocityIsInward(const G4LatticePhysical* lattice,
                                      G4int mode,
                                      const G4ThreeVector& waveVector,
                                      const G4ThreeVector& surfNorm) {
-  G4ThreeVector vDir = lattice->MapKtoVDir(mode, waveVector);
+  // Get touchable for coordinate rotations
+  const G4VTouchable* touchable = GetCurrentTouchable();
+
+  if (!touchable) {
+    G4Exception("G4CMP::PhononVelocityIsInward", "G4CMPUtils001",
+		FatalException, "Current track does not have valid touchable!");
+  }
```

Comment:
Insert a ```return false;``` line inside the if block.  If we change the exception severity in the future, the code could print the error and then continue, so we don't want a subsequent segfault.